services/galley/test/integration/API.hs

```diff
@@ -244,44 +243,44 @@ postCryptoMessage1 = do
     assertTrue_ (eqMismatch [(eve, Set.singleton ec)] [] [] . responseJsonUnsafe)
   -- Complete
   WS.bracketR2 c bob eve $ \(wsB, wsE) -> do
-    let m2 = [(bob, bc, "ciphertext2"), (eve, ec, "ciphertext2")]
+    let m2 = [(bob, bc, toBase64Text "ciphertext2"), (eve, ec, toBase64Text "ciphertext2")]
```

Comment:
Is there a backwards-incompatibility regression in the sense that before non-base64 encoded strings were accepted, which is now no longer the case, and only base64encoded test is allowed?
It's not generally a problem to accept only that, but this change/refactor isn't explained in a comment in the code nor PR description so it might be good to add that.

Reply:
I assumed it was already the case that only base64 encoded text is allowed here. The reason for the present change is just that the type for the message payload was changed from a base64-encoded text to a raw ByteString. Since now we are implementing the old API in terms of the new, we need to convert it back to a ByteString, only to be re-encoded into base64 when creating the events. Therefore, the payload needs to be valid base64 (or least be able to roundtrip back to itself after being decoded as base64) for these tests to work. They used to work before because nothing in the code was actually looking into the message at all, but they were still somewhat incorrect, because the message had to be valid base64 for the events to be parseable by clients.

Reply:
> I assumed it was already the case that only base64 encoded text is allowed here. The reason for the present change is just that the type for the message payload was changed from a base64-encoded text to a raw ByteString. Since now we are implementing the old API in terms of the new, we need to convert it back to a ByteString, only to be re-encoded into base64 when creating the events. Therefore, the payload needs to be valid base64 (or least be able to roundtrip back to itself after being decoded as base64) for these tests to work. They used to work before because nothing in the code was actually looking into the message at all, but they were still somewhat incorrect, because the message had to be valid base64 for the events to be parseable by clients.

Ah, makes sense, thanks for explaining!